src/ansys/mechanical/core/run.py

```diff
@@ -110,6 +110,10 @@ def _cli_impl(
         if input_script:
             raise Exception("Cannot open in server mode with an input script.")
 
+    # If the input_script and port are missing in batch mode, raise an exception
+    if (not graphical) and (input_script is None) and (port is None):
```

Comment:
input script `or` port, not `and`

Reply:
@koubaa if I change it to `if (not graphical) and (input_script is None) or (port is None):`, then it'll raise the exception even when port or input_script is provided. Is that what you want to happen?

Reply:
@koubaa I just added unit tests to cover those cases. I also had to add the port argument to other batch cli tests so that the exception wouldn't get raised